### ISSUE TITLE
Circular require considered harmful

### ISSUE THREAD

**Issue by taylorthurlow** [gsamokovarov/break#5]
Hi, I'm using `require "break"` in a test helper file but it's causing the following warning (not exception):

```
/Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break.rb:12: warning: /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break.rb:12: warning: loading in progress, circular require considered harmful - /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break/pry.rb
	from /Users/taylorthurlow/.anyenv/envs/rbenv/versions/2.7.2/bin/bundle:23:in  `<main>'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/versions/2.7.2/bin/bundle:23:in  `load'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/exe/bundle:34:in  `<top (required)>'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/friendly_errors.rb:123:in  `with_friendly_errors'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/exe/bundle:46:in  `block in <top (required)>'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli.rb:24:in  `start'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/base.rb:476:in  `start'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli.rb:30:in  `dispatch'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor.rb:399:in  `dispatch'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in  `invoke_command'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in  `run'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli.rb:476:in  `exec'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:28:in  `run'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in  `kernel_load'
	from /Users/taylorthurlow/.anyenv/envs/rbenv/gems/2.7.0/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in  `load'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/bin/rspec:23:in  `<top (required)>'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/bin/rspec:23:in  `load'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/exe/rspec:4:in  `<top (required)>'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/runner.rb:45:in  `invoke'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/runner.rb:71:in  `run'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/runner.rb:86:in  `run'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/runner.rb:102:in  `setup'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/configuration.rb:1613:in  `load_spec_files'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/configuration.rb:1613:in  `each'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/configuration.rb:1615:in  `block in load_spec_files'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/configuration.rb:2112:in  `load_file_handling_errors'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/rspec-core-3.10.1/lib/rspec/core/configuration.rb:2112:in  `load'
	from /Users/taylorthurlow/Code/some_ruby_gem/spec/lib/some_ruby_gem_spec.rb:1:in  `<top (required)>'
	from /Users/taylorthurlow/Code/some_ruby_gem/spec/lib/some_ruby_gem_spec.rb:1:in  `require'
	from /Users/taylorthurlow/Code/some_ruby_gem/spec/gem_helper.rb:1:in  `<top (required)>'
	from /Users/taylorthurlow/Code/some_ruby_gem/spec/gem_helper.rb:1:in  `require'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break/pry.rb:5:in  `<top (required)>'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break/pry.rb:5:in  `require'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break.rb:12:in  `<top (required)>'
	from /Users/taylorthurlow/Code/some_ruby_gem/vendor/ruby/2.7.0/gems/break-0.21.0/lib/break.rb:12:in  `require_relative'
```

Removing the additional `require "break"` from `break/pry.rb` on line 5 resolves the issue. I'm not sure if it makes a difference, but I'm using `zeitwerk` for code loading. I'm not sure if this is related though, as the warning posted above is a plain Ruby thing. Also it doesn't show up in the stack trace.

**Comment by gsamokovarov** [comment]
Thanks for the report will look this up.

**Comment by gsamokovarov** [comment]
Should be fixed in `0.30.0`. Sorry for taking so long to respond and feel free to reopen the issue if the problem persists.
